### PR TITLE
[SPARK-48543][SS] Track state row validation failures using explicit error class

### PR DESCRIPTION
**Pull request by anishshri-db**
### What changes were proposed in this pull request?
Track state row validation failures using explicit error class

### Why are the changes needed?
We want to track these exceptions explicitly since they could be indicative of underlying corruptions/data loss scenarios.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing unit tests

```
13:06:32.803 INFO org.apache.spark.util.ShutdownHookManager: Deleting directory /Users/anish.shrigondekar/spark/spark/target/tmp/spark-6d90d3f3-0f37-48b8-8506-a8cdee3d25d7
[info] Run completed in 9 seconds, 861 milliseconds.
[info] Total number of tests run: 4
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 4, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
```

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by anishshri-db** [comment]
cc - @HeartSaVioR - PTAL, thx !

**Comment by anishshri-db** [comment]
Tests are all passing. Not sure why the page is not updated.

**Review by HeartSaVioR — APPROVED** [review]
+1

**Comment by HeartSaVioR** [comment]
Wait... @anishshri-db if we consider this as worth monitoring, is it sufficient that we categorize this as internal class with state store? Wouldn't we like to give an explicit error class instead?

**Comment by anishshri-db** [comment]
@HeartSaVioR - PTAL now, thx !

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingStateStoreFormatCompatibilitySuite.scala [hunk 1]
```diff
@@ -254,7 +253,7 @@ class StreamingStateStoreFormatCompatibilitySuite extends StreamTest {
   private def findStateSchemaException(exc: Throwable): Boolean = {
     exc match {
       case _: SparkUnsupportedOperationException => true
-      case _: InvalidUnsafeRowException => true
+      case _: SparkException => true
```
Comment (HeartSaVioR) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingStateStoreFormatCompatibilitySuite.scala:256]: Can we use StateStoreKeyRowFormatValidationFailure and StateStoreValueRowFormatValidationFailure here?

**Review by HeartSaVioR — COMMENTED** [review]
Only one minor. Thanks for making change!

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingStateStoreFormatCompatibilitySuite.scala (hunk 1 above)
Comment (anishshri-db) [reply sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingStateStoreFormatCompatibilitySuite.scala]: Done

**Review by HeartSaVioR — APPROVED** [review]
+1 pending CI

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Pull request closed without merging**